controller/test/frame_detector/frame_detector_test.cpp

```diff
@@ -0,0 +1,144 @@
+#include "frame_detector.h"
+#include "proto_traits.h"
+#include "soft_rx_buffer.h"
+#include "gtest/gtest.h"
+
+constexpr uint32_t LEN = 20;
+
+constexpr FrameDetector<SoftRxBuffer<LEN>, LEN>::State STATE_LOST =
+    FrameDetector<SoftRxBuffer<LEN>, LEN>::State::LOST;
+constexpr FrameDetector<SoftRxBuffer<LEN>, LEN>::State STATE_WAIT_START =
+    FrameDetector<SoftRxBuffer<LEN>, LEN>::State::WAIT_START;
+constexpr FrameDetector<SoftRxBuffer<LEN>, LEN>::State STATE_RX_FRAME =
+    FrameDetector<SoftRxBuffer<LEN>, LEN>::State::RX_FRAME;
+
+constexpr uint8_t MARK = static_cast<uint8_t>('.');
+
+SoftRxBuffer<LEN> rx_buf('.');
```

Comment:
Use MARK directly?

Reply:
This wasn't resolved

Reply:
It's easier and more clear to the eye writing test cases like `".aaa."` instead of `"\xe2aaaa\xe2"`

Reply:
Doh! Fixed